vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -138,12 +138,19 @@ def _get_parametrized_config(target: ModelID, ctd_parameters: List[CallbackTrigg
     config["data_frame"] = {}
 
     for ctd in ctd_parameters:
-        selector_value = ctd[
-            "value"
-        ]  # TODO: needs to be refactored so that it is independent of implementation details
+        # TODO: needs to be refactored so that it is independent of implementation details
+        selector_value = ctd["value"]
+
         if hasattr(selector_value, "__iter__") and ALL_OPTION in selector_value:  # type: ignore[operator]
             selector: SelectorType = model_manager[ctd["id"]]
-            selector_value = selector.options
+
+            # Even if options are provided as List[Dict], the Dash component only returns a List of values.
+            # So we need to ensure that we always return a List only as well to provide consistent types.
+            if all(isinstance(i, dict) for i in selector.options):
+                selector_value = [value["value"] for value in selector.options]
+            else:
+                selector_value = selector.options
+
         selector_value = _validate_selector_value_none(selector_value)
```

Comment:
what are the possible types of `selector_value` into function `_validate_selector_value_none`? Would it only be `NONE_OPTION` or `list` now?

Reply:
No, it can also be a single value. The typing of that function is actually correct now, so  

`def _validate_selector_value_none(value: Union[SingleValueType, MultiValueType]) -> ValidatedNoneValueType:`

where:

```
SingleValueType = Union[StrictBool, float, str, date]
MultiValueType = Union[List[StrictBool], List[float], List[str], List[date]]
```